doc/design/record_fetcher_proposal.md

```diff
@@ -0,0 +1,178 @@
+documents the manifest describes.
+
+## Objective
+
+Define a system for tracking the associations between the various crude scrapings that describe a record. How documents
+are fetched and how software is deployed are both directly relevant to this space, but are out of scope for this
+document.
+
+## Background
+
+Record collection like we're trying to do is a dumpster fire of inconsistency. In some cases the fields we want are in a
+single document, but frequently the fields we require for a single logical record span multiple documents. Rather than
+having every scraper developer wrestle with this complexity, present design discussions are leaning toward separate
+source fetching and field extraction stages. This allows us to build a corpus of crude source material which may be more
+easily refined offline, with greatly reduced risk of the extraction process tainting the original data.
+
+Separately, there is a large concern surrounding tracking of data lineage to prevent (or at least reliably detect)
+accidental or intentional tampering of data. Offline processing of the raw source data makes it possible to cheaply
+iterate on extraction tooling and forward propagate the results, while also being able to compare against previous
+extractions, without the need for querying a service with questionable availability and durability.
+
+## Requirements
+
+We need a way to store raw information pertaining to logical records. Given logical records can span documents, we also
+need to store a representation (a manifest) of those relationships. Given records are capable of changing over time, we
+need to be able to track distinct snapshots over time. Additionally, we need a service which will take this information
+and persist it in a meaningful way. Analysis of this information is not in scope.
+
+### Manifests
+
+A manifest needs to be able to represent the raw materials associated with a record. The minimal identifiers for the
+record itself are:
+* a unique ID
+* which region and record type the manifest represents (e.g., "USA/FL/Bay/Court")
+* some sort of case identifier
+* the timestamp of collection
+
+A bunch of record-identifying metadata is cool, but we also need to be able to represent the documents that compose the
+record. To that end, we'll require a list of document references. Each reference will contain:
+* a content type (we can just use [MIME
+types](https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types/Common_types) here)
+* a digest of the content
+* a URI for where that document's contents are durably stored in PDAP infrastructure
+  * this isn't necessary if we adopt storage infrastructure that allows us to reliably derive this from the digest
+* an optional comment expressing where this specific document came from
+
+In order for us to trust our collected record results, we need to be able to verify that we trust the software which
```

Comment:
Not disagreeing, but I hope we'll be able to come up with some per-language libraries to help with this, it could be daunting for the average person

Reply:
Oh, I totally agree and that is the intention.